tutorials/manage-instances-with-terraform-and-functions/index.mdx

```diff
@@ -0,0 +1,311 @@
+            skip_credentials_validation = true
+            skip_region_validation      = true
+        }
+        }
+        /*
+        For the credentials part:
+        ==> Create a ~/.aws/credentials:
+        [default]
+        aws_access_key_id=<SCW_ACCESS_KEY>
+        aws_secret_access_key=<SCW_SECRET_KEY>
+        region=fr-par
+        */
+    ```
+
+5. Edit the `provider.tf` file and add Scaleway as a provider:
+    ```json
+    terraform {
+      required_providers {
+        scaleway = {
+          source  = "scaleway/scaleway"
+          version = "2.2.7"
+        }
+      }
+      required_version = ">= 0.13"
+    }
+    ```
+
+  6. Specify the following variables in the `variables.tf` file
+    ```json
+    variable "zone" {
+      type = string
+    }
+
+    variable "region" {
+      type = string
+    }
+
+    variable "env" {
+      type = string
+    }
+
+    variable "project_id" {
+      type        = string
+      description = "Your project ID"
+    }
+
+    variable "auth_token" {
+        type = string
+        description = "Scaleway authtentication token used in the function"
+    }
+    ```
+
+6. Add the variable values to `terraform.tfvars` 
+    ```bash
+    zone                           = "fr-par-1"
+    region                         = "fr-par"
+    env                            = "dev"
+    project_id                     = "Your project ID"
+    auth_token                     = "Your Scaleway Secret Key"
+    ```
+
+
+## Writing the code
+
+For this example we will use the native python library `urllib`, which allows us to keep the package slim.
+
+<Message type="tip">
+
+All information about the Instance API can be found in the [developers documentation](https://developers.scaleway.com/en/products/instance/api/#get-2c1c6f).
+
+</Message>
+
+
+Inside the `function` folder create a `handler.py` file with following contents:
+  ```py
+  import os
+  from urllib import request, parse, error
+  import json
+
+  auth_token = os.environ["X-AUTH-TOKEN"]
+
+  def handle(event, context):
+      ## get information from cron
+      event_body = eval(event["body"])
+      zone = event_body["zone"]
+      server_id = event_body["server_id"]
+      action = event_body["action"]  # action should be "poweron" or "poweroff"
+
+      # create request
+      url = "https://api.scaleway.com/instance/v1/zones/{zone}/servers/{server_id}/action"
+      data = json.dumps({"action": action}).encode(` ascii `)
+      req = request.Request(url, data=data, method="POST")
+      req.add_header(` Content - Type `, ` application / json `)
+      req.add_header(` X - Auth - Token `, auth_token)
+
+      # Sending request to Instance API
+      try:
+          res = request.urlopen(req).read().decode()
+      except error.HTTPError as e:
+          res = e.read().decode()
+
+      return {
+          "body": json.loads(res),
+          "statusCode": 200,
+      }
+  ```
```

Comment:
```suggestion
```python3
import os
from urllib import request, parse, error
import json

auth_token = os.environ["X-AUTH-TOKEN"]


def handle(event, context):
    ## get information from cron
    event_body = eval(event["body"])
    zone = event_body["zone"]
    server_id = event_body["server_id"]
    action = event_body["action"]  # action should be "poweron" or "poweroff"

    # create request
    url = "https://api.scaleway.com/instance/v1/zones/{zone}/servers/{server_id}/action"
    data = json.dumps({"action": action}).encode(` ascii `)
    req = request.Request(url, data=data, method="POST")
    req.add_header(` Content - Type `, ` application / json `)
    req.add_header(` X - Auth - Token `, auth_token)

    # Sending request to Instance API
    try:
        res = request.urlopen(req).read().decode()
    except error.HTTPError as e:
        res = e.read().decode()

    return {
        "body": json.loads(res),
        "statusCode": 200,
    }

```
```

Reply:
based on @thomas-tacquet review